### PR TITLE
docs: Improvements to two READMEs in /test

### PR DESCRIPTION
**Pull request by merland**
**Before** 
- Most commands in test/README.md are relative to the repo root, but not all.
- The locations of the log files are not described. 

**After** 
- All commands are relative to the repo root.
- Info about the location of the test framework log files has been added.

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -108,17 +108,23 @@ tests will fail. If this happens, remove the cache directory (and make
...
+ * `<test data directory>/<node>/regtest/debug.log`. 
+ 
+The `<test data directory>` path is by default set to the temporary directory of your OS, and its full path is written as the very first line of the test output. The `<node>` part identifies the relevant test node (from `node1` to `node4`).  
+
 The tests contain logging at different levels (debug, info, warning, etc). By
 default:
 
 - when run through the test_runner harness, *all* logs are written to
-  `test_framework.log` and no logs are output to the console.
+  test_framework.log` and no logs are output to the console.
```
Comment (practicalswift) [inline test/README.md]: This change wasn't intentional? :-)

**Comment by practicalswift** [comment]
Concept ACK

Thanks for improving the documentation!

File: test/README.md [hunk 2]
```diff
@@ -108,17 +108,23 @@ tests will fail. If this happens, remove the cache directory (and make
...
 killall bitcoind
 ```
 
 ##### Test logging
 
+The test framework logs to two files: 
+ * `<test data directory>/test_framework.log` 
+ * `<test data directory>/<node>/regtest/debug.log`. 
+ 
+The `<test data directory>` path is by default set to the temporary directory of your OS, and its full path is written as the very first line of the test output. The `<node>` part identifies the relevant test node (from `node1` to `node4`).
```
Comment (jonatack) [inline test/README.md]: Would be nice to stay within the max line length seen in the file.

**Review by jonatack — COMMENTED** [review]
Concept ACK. Filenames from root seems clearer to me too, though I don't have a strong opinion. Note that in test/functional/README there are non-root filenames as well, so if consensus is in that direction why not update both.

File: test/README.md (hunk 1 above)
Comment (merland) [reply test/README.md]: Indeed! Fixed.

File: test/README.md (hunk 2 above)
Comment (merland) [reply test/README.md]: Agreed! Fixed.

File: test/README.md [hunk 3]
```diff
@@ -108,17 +108,23 @@ tests will fail. If this happens, remove the cache directory (and make
...
-rm -rf cache
+rm -rf test/cache
 killall bitcoind
 ```
 
 ##### Test logging
 
+The test framework logs to two files: 
+ * `<test data directory>/test_framework.log` 
+ * `<test data directory>/<node>/regtest/debug.log`.
```
[the remark this answers is not in the document]
Comment (merland) [reply test/README.md]: Good idea! Made it so.

**Comment by merland** [comment]
> Note that in test/functional/README there are non-root filenames as well, so if consensus is in that direction why not update both.

Good find. I'll try do make it happen, but I'd rather do it in another PR. Keeping things small.

File: test/README.md [hunk 4]
```diff
@@ -108,12 +108,20 @@ tests will fail. If this happens, remove the cache directory (and make
...
 ```
 
 ##### Test logging
 
+The test framework logs to two files: 
+ * `<test data directory>/test_framework.log` 
+ * `<test data directory>/node<number>/regtest/debug.log`. 
+ 
+The `<test data directory>` path is by default set to the temporary directory of
+your OS, and its full path is written as the very first line of the test output.
```
Comment (MarcoFalke) [inline test/README.md]: It is not of the OS, but rather of the python environment. And it can be overwritten via a parameter to the test (runner)

**Review by maflcko — COMMENTED** [review]
ACK, but I agree that both files should be fixed up. I think 12 lines to review is small enough to include more changes.

File: test/README.md [hunk 5]
```diff
@@ -108,12 +108,20 @@ tests will fail. If this happens, remove the cache directory (and make
...
+The `node<number>` part identifies the relevant test node (from `node1` to`node4`).
```
Comment (jonatack) [inline test/README.md]: Add a space after "to" here: to\`node4\`

File: test/README.md (hunk 4 above)
Comment (jonatack) [inline test/README.md]: Suggestion: "its full path is provided in the first line of the test output."

Comment (merland) [reply test/README.md]: Well, the python tempfile module is used to achieve cross-platform support, but the current phrasing "...is by default set to the temporary directory of your OS" is still valid, IMO.   

Regarding the overwriting ability: My thinking was that adding  "by default" would suggest to the reader that there are other options (which are mentioned elsewhere). Brevity is nice but I could add more info if you think it is needed.

File: test/README.md (hunk 5 above)
Comment (merland) [reply test/README.md]: > ACK, but I agree that both files should be fixed up. I think 12 lines to review is small enough to include more changes.

@MarcoFalke Fair enough. I'll look into it!

Comment (merland) [reply test/README.md]: @jonatack Thanks, fixed! (blush)

File: test/README.md (hunk 4 above)
Comment (merland) [reply test/README.md]: > Suggestion: "its full path is provided in the first line of the test output."

Agreed, better! Fixed.

Comment (MarcoFalke) [reply test/README.md]: Yeah, no strong opinion. Just for reference: https://docs.python.org/3.8/library/tempfile.html#tempfile.gettempdir

File: test/functional/README.md [hunk 6]
```diff
@@ -82,7 +82,7 @@ P2P messages. These can be found in the following source files:
 
 #### Using the P2P interface
 
-- `messages.py` contains all the definitions for objects that pass
+- [messages.py](test/functional/test_framework/messages.py) contains all the definitions for objects that pass
```
Comment (jonatack) [inline test/functional/README.md]: Are the label and link inverted here?

File: test/functional/README.md [hunk 7]
```diff
@@ -96,32 +96,33 @@ the Bitcoin Core node application logic. For custom behaviour, subclass the
 P2PInterface object and override the callback methods.
 
 - Can be used to write tests where specific P2P protocol behavior is tested.
-Examples tests are `p2p_unrequested_blocks.py`, `p2p_compactblocks.py`.
+Examples tests are [p2p_unrequested_blocks.py](p2p_unrequested_blocks.py), [p2p_compactblocks.py](p2p_compactblocks.py).
 
-### test-framework modules
+### Test framework modules
+The following are some useful modules for test developers. They are all located in [test/functional/test_framework](test_framework/)
```
Comment (jonatack) [inline test/functional/README.md]: Missing punctuation at end of the line. For brevity can omit "some" and "all". Would it be better to add a slash at the end of the `test/functional/test_framework/` label?

Comment (jonatack) [reply test/functional/README.md]: (Ideally limit line lengths to 120 chars)

File: test/functional/README.md (hunk 6 above)
Comment (jonatack) [reply test/functional/README.md]: (the contents)

Comment (merland) [reply test/functional/README.md]: No, but the link was a bit off. Fix coming shortly.

File: test/functional/README.md (hunk 7 above)
Comment (merland) [reply test/functional/README.md]: Thanks for your feedback. Fixed.

File: test/functional/README.md [hunk 8]
```diff
@@ -4,13 +4,13 @@
 
 #### Example test
 
-The [example_test.py](example_test.py) is a heavily commented example of a test case that uses both
-the RPC and P2P interfaces. If you are writing your first test, copy that file
-and modify to fit your needs.
+The python script [test/functional/example_test.py](example_test.py) is a heavily commented example
```
[the remark this answers is not in the document]
Comment (merland) [reply test/functional/README.md]: Fixed, thanks.

File: test/README.md [hunk 9]
```diff
@@ -108,12 +108,20 @@ tests will fail. If this happens, remove the cache directory (and make
...
 killall bitcoind
 ```
 
 ##### Test logging
 
+The test framework logs to two files: 
+ * `<test data directory>/test_framework.log` 
+ * `<test data directory>/node<number>/regtest/debug.log`. 
+ 
+The `<test data directory>` path is by default set to the temporary directory of
```
[the remark this answers is not in the document]
Comment (merland) [reply test/README.md]: Brevity is key, but personally I think your suggestion would be taking it too far. IMO most readers would be interested in the OS part. 
Anyone else?

**Comment by merland** [comment]
squashed

File: test/README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (merland) [reply test/README.md]: @jonatack I went with your suggestion here (with an added comma), due to opinion not strong enough :)

**Comment by merland** [comment]
All feedback has been adressed, can this PR be merged?

**Comment by MarcoFalke** [comment]
The tests don't pass

**Comment by merland** [comment]
Missed the little red X again.... Fixed some linting errors.

**Comment by merland** [comment]
All feedback has been adressed, can this PR be merged?

File: test/README.md [hunk 10]
```diff
@@ -108,12 +108,19 @@ tests will fail. If this happens, remove the cache directory (and make
...
 ```
 
 ##### Test logging
 
+The test framework logs to two files:
+ * `<test data directory>/test_framework.log`
+ * `<test data directory>/node<number>/regtest/debug.log`.
+
+The first line of the test output provides the test data directory path. The
+node number is the relevant test node, from node1 to node4.
```
Comment (MarcoFalke) [inline test/README.md]: tests can set their own number of nodes and it starts at node0, btw

**Comment by merland** [comment]
I decided to close this PR. Thanks for all the good feedback but I feel it grew too much. Not in number of lines but in number of comments and elapsed calendar time. 
If someone wants to use these ideas in another PR, please feel free to do so! (Or reopen this one.)

**Pull request closed without merging**